microsoft/testsuites/asr/cvt.py

```diff
@@ -0,0 +1,275 @@
+
+        extension_instance_view = extension.get(
+            name="Linux",
+        ).instance_view
+
+        extension_substatus = ""
+        for substatus in extension_instance_view.substatuses:
+            log.info(f"Substatus : '{substatus}'")
+            code = substatus.code
+            if code is not None and code.find("platform") != -1:
+                extension_substatus = code
+                break
+        if not extension_substatus:
+            return extension_substatus
+
+        log.info(f"Substatus : '{extension_substatus}'")
+        split_status = re.split(",|/", extension_substatus)
+        for component_status in split_status:
+            if (
+                component_status is not None
+                and component_status.find("osidentifier") != -1
+            ):
+                log.info(f"Component status : '{component_status}'")
+                os = base64.b64decode(
+                    re.split(":", component_status)[-1]
+                ).decode()
+
+        log.info(f"os : '{os}'")
+        return os
+
+    def install_asr_extension_distro(
+        self, log: Logger, node: Node, os: str
+    ) -> None:
+        task_id = str(uuid.uuid4())
+        extension_publishers = {
+            "SLES11-SP3-64",
+            "SLES11-SP4-64",
+            "RHEL6-64",
+            "RHEL7-64",
+            "UBUNTU-14.04-64",
+            "UBUNTU-16.04-64",
+            "OL6-64",
+            "OL7-64",
+        }
+        extension_test = {
+            "SLES11-SP3-64",
+            "SLES11-SP4-64",
+            "OL6-64",
+            "RHEL7-64",
+        }
+        publisher_name = "Microsoft.Azure.SiteRecovery.Test"
+        extension_name = self.get_extension_name(os=os, log=log)
+        if os in extension_publishers:
+            if os in extension_test:
+                extension_name = extension_name + "Test"
+        else:
+            publisher_name = "Microsoft.Azure.SiteRecovery2.Test"
+
+        cur_time = datetime.datetime.now().isoformat() + "Z"
+        settings = {
+            "publicObject": "",
+            "module": "a2a",
+            "timeStamp": cur_time,
+            "commandToExecute": "Install",
+            "taskId": task_id,
+        }
+        extension = node.features[AzureExtension]
+
+        result = extension.create_or_update(
+            name=extension_name,
+            publisher=publisher_name,
+            type_=extension_name,
+            type_handler_version="1.0",
+            auto_upgrade_minor_version=True,
+            settings=settings,
+        )
+        assert_that(result["provisioning_state"]).described_as(
+            "Expected the extension to succeed"
+        ).is_equal_to("Succeeded")
+
+    def run_script(
+        self, node: Node, log: Logger, test_dir: str
+    ) -> ExecutableResult:
+        timer = create_timer()
+        script: CustomScript = node.tools[self._cvt_script]
+        result = script.run(parameters=test_dir, timeout=19800, sudo=True)
```

Comment:
test case timeout is 12000 is less than script execute time out value 19800, then the case may fail for timeout